lib/charms/layer/apache_bigtop_base.py

```diff
@@ -53,9 +53,14 @@ def get_ip_for_interface(self, network_interface):
 
         @param str network_interface: either the name of the
         interface, or a CIDR range, in which we expect the interface's
-        ip to fall.
+        ip to fall. Also accepts 0.0.0.0 as a special case, which will
+        simply return 0.0.0.0.
```

Comment:
Couple things:

- someone may enter `0.0.0.0/0` or `0/0` as a shortcut.  can we support those in the special case handler?  perhaps it's enough to test if `network_interface` starts with a `0`.
- how about `::` in case someone only has ipv6 and wants to reset to listening on all interfaces?  do kafka/zk properties treat `::` as "all ipv6 interfaces"?